contextualized/regression/tests.py

```diff
@@ -47,44 +48,122 @@ def setUp(self):
         self.c_dim, self.x_dim, self.y_dim = c_dim, x_dim, y_dim
         self.C, self.X, self.Y = C.numpy(), X.numpy(), Y.numpy()
 
-    def _quicktest(self, model, univariate=False, correlation=False, markov=False):
-        """
+    def _get_model(
+        self,
+        model,
+        dataclass_type,
+        pred_dl_type,
+        correlation=False,
+        univariate=False,
+        markov=False,
+    ):
 
-        :param model:
-        :param univariate:  (Default value = False)
-        :param correlation:  (Default value = False)
-        :param markov:  (Default value = False)
+        get_dataclass = {
+            "datamodule": lambda x, **kwargs: x.datamodule(**kwargs),
+            "dataloader": lambda x, **kwargs: x.dataloader(**kwargs),
+        }
 
-        """
-        print(f"\n{type(model)} quicktest")
+        # get subset of data to use for y_true
+        def get_xy_pred(dc):
```

Comment:
Tests should align with our usecases, and the datamodules main function is to hide all the individual datasets from the user. We should probably use the dataset kwarg in the trainer to specify prediction data, rather than adding functions like `get_xy_pred` and `_get_model`.